Add ability to add custom packages

crees opened the pull request:
Since drm-kmod is never going to match 12.2, we should just fix this now.

Any line starting with https in the package files is pulled separately.
They should be in reverse order of dependencies.

I've build drm-kmod for 12.2, but of course these could come from anywhere.

The logic for pulling pkgs from https is reusable and should work just fine with any location- I fully understand that you may not want it to pull from some untrusted and unknown site!

crees committed:
Add ability to add custom packages

Since drm-kmod is never going to match 12.2, we should just fix this now.

Any line starting with https in the package files is pulled separately.
They should be in reverse order of dependencies.

I've build drm-kmod for 12.2, but of course these could come from anywhere.

crees commented:
Er, don't expect this to work with 12.1-release btw-- it'll probably build just fine, but it might end up crashing on use.  This fix is for 12.2-- I understand it was the only blocker?  You appear to have moved on anyway, so this would solve it.

probonopd commented:
Thank you very much @crees. As outlined in https://github.com/helloSystem/ISO/issues/1#issuecomment-787102817 I'd like to put this on hold until we find a way to produce packages on Cirrus CI.

crees committed:
Only fetch experimental packages from cirrus

These are trusted, as you can refer back to the github source

Cache these in the usual place

Currently supported for 12.1 and 12.2; they are built from
https://github.com/crees/drm-kmod-FreeBSD-12,
so the .cirrus.yml would need updating for further versions

crees commented:
I haven't run tested this one yet-- will let you know.

probonopd commented:
`Certificate verification failed for /CN=api.cirrus-ci.com`... why?

crees commented:
pkg appears to be fussy about https certificates it would appear.  I've instead used fetch to download them to the cache dir and install them that way- this has the advantage that they don't need refetching.

This does give me the proper lovely looking desktop, but slim doesn't automatically log me in, I end up with slim asking me to log in.  "liveuser" is pre-filled, so i only need to hit "Enter", but that doesn't seem right to me.  Is that what sometimes happens?  I don't see how this change would cause that.

probonopd commented:
Thanks @crees, is there any way to do this but not break 12.1 builds?
Possibly we need to make separate packages files for 12.1, 12.2, 13.0?

crees commented:
Sure, we can do this.  I'll look at making a package for both.

crees commented:
We don't need to do for 13.0- drm-kmod from the packages will be fine.  Let's see if the test works...

probonopd commented:
Fingers crossed!
